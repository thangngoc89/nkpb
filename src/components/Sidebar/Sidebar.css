.wrap {
  width: 100vw;
  left: 0;
  height: 100vh;
  top: 0;
  position: absolute;
}

.iconWrap {
  position: absolute;
}

.icon {
  float: left;
  margin-top: 60px;
  margin-left: 40px;
  color: #fff;
  cursor: pointer;
  width: 38px;
  z-index: 1;
  padding: 0;
}

.titleWrap {
  margin: 60px 40px 20px 40px;
  width: 100%;
  height: calc(100% - 60px - 20px);
}

.title {
  font-size: 34px;
  font-weight: 400;
  color: #fff;
  z-index: 1;
  position: absolute;
  line-height: 40px;
  margin-left: 52px;
}

.titleNoIcon {
  margin-left: 10px;
}

.hide {
  display: none !important;
}

.panelWrap {
  width: 100vw;
  left: 0;
  height: 100vh;
  top: 0;
  position: absolute;
  box-sizing: border-box;
}

.leftPanel {
  width: 300px;
  height: 100%;
  background-color: #3f51b5;
  float: left;

  & paper-item {
    padding: 0 0 0 50px;
    display: flex;
    justify-content: flex-start;

    font-size: 18px;
    font-weight: 300;
    padding-top: 5px;
    padding-bottom: 5px;
    position: relative;
    text-align: right;
    cursor: pointer;
    height: 50px;
    text-transform: uppercase;
    line-height: 25px;

    &:hover {
      background-color: rgba(0, 0, 0, 0.1);
    }

    & iron-icon {
      margin-right: 20px;
    }
  }

  & paper-listbox {
    margin-top: 120px;
    position: relative;
    background-color: transparent;
    color: #fff;
  }
}

.rightPanel {
  width: calc(100% - 300px);
  height: 100%;
  float: right;
  padding-left: 40px;
  padding-right: 40px;
  padding-top: 80px;
  box-sizing: border-box;
}
