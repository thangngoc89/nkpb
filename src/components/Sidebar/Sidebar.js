import React, { PropTypes } from 'react'
import { Link } from 'react-router'
import cx from 'classnames'
import styles from './Sidebar.css'

const List = ({ list }) => {
  const tabStyles = `
    .paper-item-0 {
      --paper-item-selected-weight: 300 !important;
      --paper-item-selected: {
        background: rgba(0, 0, 0, 0.3);
      }
      --paper-item-focused-before: {
        opacity: 0;
      }
    };
  `

  return (
    <div className={styles.leftPanel}>
      <style is="custom-style" dangerouslySetInnerHTML={{ __html: tabStyles }} />
      <paper-listbox>
        {
          list.map((item) => (
            <paper-item key={item.name}>
              <iron-icon icon={item.icon} />
              {item.name}
              <paper-ripple />
            </paper-item>
          ))
        }
      </paper-listbox>
    </div>
  )
}

List.propTypes = {
  list: PropTypes.array.isRequired,
}

const Sidebar = ({ title, icon, list, children }) => (
  <div className={styles.wrap}>
    <Link
      to="/"
      className={cx(styles.iconWrap, {
        [styles.hide]: !icon.display,
      })}
    >
      <paper-icon-button className={styles.icon} icon={icon.icon} />
    </Link>
    <div className={styles.titleWrap}>
      <h1
        className={cx(styles.title, {
          [styles.titleNoIcon]: !icon.display,
        })}
      >
        { title }
      </h1>
    </div>

    <div className={styles.panelWrap}>
      <List list={list} />
      <div className={styles.rightPanel}>
        { children }
      </div>
    </div>
  </div>
)

Sidebar.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  list: PropTypes.array.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ])
}

export default Sidebar
