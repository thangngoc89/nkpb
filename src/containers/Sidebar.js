import { connect } from 'react-redux'
import Sidebar from '../components/Sidebar'


const mapStateToProps = ({ sidebar }) => ({ ...sidebar })

export default connect(
  mapStateToProps,
  // (dispatch) => bindActionCreators(sidebarActions, dispatch),
)(Sidebar)
