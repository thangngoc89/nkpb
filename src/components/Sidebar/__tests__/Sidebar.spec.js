import React from 'react'
import renderer from 'react-test-renderer'
import Sidebar from '../Sidebar.js'

const props = {
  title: 'Phước Bình',
  icon: {
    display: false,
    icon: 'arrow-back',
    link: '/',
  },
  list: [
    {
      name: 'Khách hàng',
      icon: 'social:person',
    },
    {
      name: 'Plugins',
      icon: 'device:widgets',
    },
    {
      name: 'About',
      icon: 'info-outline',
    },
  ],
}

describe('Sidebar', () => {
  it('should not require children to render', () => {
    const component = renderer.create(
      <Sidebar {...props} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with no icon', () => {
    const component = renderer.create(
      <Sidebar {...props} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with icon', () => {
    const newProps = {
      ...props,
      icon: {
        ...props.icon,
        display: true,
      },
    }

    const component = renderer.create(
      <Sidebar {...newProps} />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
