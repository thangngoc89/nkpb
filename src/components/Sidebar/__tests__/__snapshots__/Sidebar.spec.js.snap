exports[`Sidebar should not require children to render 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1
      className="undefined">
      Phước Bình
    </h1>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
              .paper-item-0 {
                --paper-item-selected-weight: 300 !important;
                --paper-item-selected: {
                  background: rgba(0, 0, 0, 0.3);
                }
                --paper-item-focused-before: {
                  opacity: 0;
                }
              };
            "
          }
        }
        is="custom-style" />
      <paper-listbox>
        <paper-item>
          <iron-icon
            icon="social:person" />
          Khách hàng
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="device:widgets" />
          Plugins
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="info-outline" />
          About
          <paper-ripple />
        </paper-item>
      </paper-listbox>
    </div>
    <div
      className={undefined} />
  </div>
</div>
`;

exports[`Sidebar should render 1 element as a child 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1
      className="undefined">
      Phước Bình
    </h1>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
              .paper-item-0 {
                --paper-item-selected-weight: 300 !important;
                --paper-item-selected: {
                  background: rgba(0, 0, 0, 0.3);
                }
                --paper-item-focused-before: {
                  opacity: 0;
                }
              };
            "
          }
        }
        is="custom-style" />
      <paper-listbox>
        <paper-item>
          <iron-icon
            icon="social:person" />
          Khách hàng
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="device:widgets" />
          Plugins
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="info-outline" />
          About
          <paper-ripple />
        </paper-item>
      </paper-listbox>
    </div>
    <div
      className={undefined}>
      <p>
        Whatever
      </p>
    </div>
  </div>
</div>
`;

exports[`Sidebar should render multi elements as children 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1
      className="undefined">
      Phước Bình
    </h1>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
              .paper-item-0 {
                --paper-item-selected-weight: 300 !important;
                --paper-item-selected: {
                  background: rgba(0, 0, 0, 0.3);
                }
                --paper-item-focused-before: {
                  opacity: 0;
                }
              };
            "
          }
        }
        is="custom-style" />
      <paper-listbox>
        <paper-item>
          <iron-icon
            icon="social:person" />
          Khách hàng
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="device:widgets" />
          Plugins
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="info-outline" />
          About
          <paper-ripple />
        </paper-item>
      </paper-listbox>
    </div>
    <div
      className={undefined}>
      <p>
        Whatever
      </p>
      <p>
        So cool
      </p>
    </div>
  </div>
</div>
`;

exports[`Sidebar should render with icon 1`] = `
<div
  className={undefined}>
  <a
    className={undefined}
    onClick={[Function bound handleClick]}
    style={Object {}}>
    <paper-icon-button
      className={undefined}
      icon="arrow-back" />
  </a>
  <div
    className={undefined}>
    <h1
      className="">
      Phước Bình
    </h1>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
              .paper-item-0 {
                --paper-item-selected-weight: 300 !important;
                --paper-item-selected: {
                  background: rgba(0, 0, 0, 0.3);
                }
                --paper-item-focused-before: {
                  opacity: 0;
                }
              };
            "
          }
        }
        is="custom-style" />
      <paper-listbox>
        <paper-item>
          <iron-icon
            icon="social:person" />
          Khách hàng
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="device:widgets" />
          Plugins
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="info-outline" />
          About
          <paper-ripple />
        </paper-item>
      </paper-listbox>
    </div>
    <div
      className={undefined} />
  </div>
</div>
`;

exports[`Sidebar should render with no icon 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1
      className="undefined">
      Phước Bình
    </h1>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
              .paper-item-0 {
                --paper-item-selected-weight: 300 !important;
                --paper-item-selected: {
                  background: rgba(0, 0, 0, 0.3);
                }
                --paper-item-focused-before: {
                  opacity: 0;
                }
              };
            "
          }
        }
        is="custom-style" />
      <paper-listbox>
        <paper-item>
          <iron-icon
            icon="social:person" />
          Khách hàng
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="device:widgets" />
          Plugins
          <paper-ripple />
        </paper-item>
        <paper-item>
          <iron-icon
            icon="info-outline" />
          About
          <paper-ripple />
        </paper-item>
      </paper-listbox>
    </div>
    <div
      className={undefined} />
  </div>
</div>
`;
